File: src/01-easy/the-river-ii-/solution.swift
func ds(_ n:Int)->Int{var x=n,s=0;while x>0{s+=x%10;x/=10};return s}
let r=Int(readLine()!)!
let lo=max(1,r-45)
print((lo..<r).contains{$0+ds($0)==r} ? "YES" : "NO")

File: src/01-easy/the-river-ii-/solution.js
const ds=n=>[...String(n)].reduce((s,c)=>s+ +c,0);
const r=+readline();
let found=false;
for(let x=Math.max(1,r-45);x<r;x++)if(x+ds(x)===r){found=true;break;}
console.log(found?"YES":"NO");
